docs/data/date-pickers/date-range-picker/ResponsiveDateRangePicker.tsx

```diff
@@ -1,49 +1,53 @@
-import { Dayjs } from 'dayjs';
-import TextField from '@mui/material/TextField';
+import dayjs from 'dayjs';
+import Typography from '@mui/material/Typography';
 import Box from '@mui/material/Box';
 import Stack from '@mui/material/Stack';
-import { LocalizationProvider } from '@mui/x-date-pickers-pro';
-import { AdapterDayjs } from '@mui/x-date-pickers-pro/AdapterDayjs';
-import { MobileDateRangePicker } from '@mui/x-date-pickers-pro/MobileDateRangePicker';
-import { DesktopDateRangePicker } from '@mui/x-date-pickers-pro/DesktopDateRangePicker';
-import { DateRange } from '@mui/x-date-pickers-pro/DateRangePicker';
+import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs';
+import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
+import { Unstable_NextDateRangePicker as NextDateRangePicker } from '@mui/x-date-pickers-pro/NextDateRangePicker';
+import { Unstable_MobileNextDateRangePicker as MobileNextDateRangePicker } from '@mui/x-date-pickers-pro/MobileNextDateRangePicker';
+import { Unstable_DesktopNextDateRangePicker as DesktopNextDateRangePicker } from '@mui/x-date-pickers-pro/DesktopNextDateRangePicker';
 
-export default function ResponsiveDateRangePicker() {
-  const [value, setValue] = React.useState<DateRange<Dayjs>>([null, null]);
+function GridItem({
+  label,
+  children,
+  spacing = 1,
+}: {
+  label: string;
+  children: React.ReactNode;
+  spacing?: number;
+}) {
+  return (
+    <Box sx={{ display: 'flex', flexDirection: 'column' }}>
+      <Typography variant="body2" sx={{ mb: spacing }}>
+        {label}
+      </Typography>
+      {children}
+    </Box>
+  );
+}
 
+export default function ResponsiveDateRangePicker() {
   return (
-    <Stack spacing={3}>
-      <LocalizationProvider dateAdapter={AdapterDayjs}>
-        <MobileDateRangePicker
-          localeText={{ start: 'Mobile start', end: 'Mobile end' }}
-          value={value}
-          onChange={(newValue) => {
-            setValue(newValue);
-          }}
-          renderInput={(startProps, endProps) => (
-            <React.Fragment>
-              <TextField {...startProps} />
-              <Box sx={{ mx: 2 }}> to </Box>
-              <TextField {...endProps} />
-            </React.Fragment>
-          )}
-        />
-        <DesktopDateRangePicker
-          localeText={{ start: 'Desktop start', end: 'Desktop end' }}
-          value={value}
-          onChange={(newValue) => {
-            setValue(newValue);
-          }}
-          renderInput={(startProps, endProps) => (
-            <React.Fragment>
-              <TextField {...startProps} />
-              <Box sx={{ mx: 2 }}> to </Box>
-              <TextField {...endProps} />
-            </React.Fragment>
-          )}
-        />
-      </LocalizationProvider>
-    </Stack>
+    <LocalizationProvider dateAdapter={AdapterDayjs}>
+      <Stack spacing={4}>
+        <GridItem label="For mobile" spacing={2}>
+          <MobileNextDateRangePicker
```

Comment:
Houston—we have a problem with mobile range picker. 😢 
Notice how nor `start`, nor `end` toolbar section receives bold style as being actively selected one.
And clicking on any of the sections yields an error...

https://user-images.githubusercontent.com/4941090/203763826-c88752d7-bbb1-4413-8ced-b993fc983b71.mov

Reply:
Seems fair. We can and should do it in a separate PR. 👌

Reply:
#6989

I will play with it before opening it for review :+1: